CRM/Upgrade/Incremental/php/FiveThirtyOne.php

```diff
@@ -74,6 +74,8 @@ public function upgrade_5_31_alpha1($rev) {
    */
   public function upgrade_5_31_0($rev) {
     $this->addTask(ts('Upgrade DB to %1: SQL', [1 => $rev]), 'runSql', $rev);
+    $this->addTask('Add is_non_case_email_skipped column to civicrm_mail_settings', 'addColumn',
+      'civicrm_mail_settings', 'is_non_case_email_skipped', "TINYINT DEFAULT 0 NOT NULL COMMENT 'Skip emails which do not have a Case ID or Case hash'");
```

Comment:
Yes it's hard to know which function it goes in - while it's still in master it would be 5_31_alpha1. When it gets branched to release-candidate it would be 5_31_beta1. The timing is difficult.

But the reason for the test fail is that one of the commits is trying to add the whole file instead of just changes. Maybe some git mixup.

Reply:
We wouldn't merge it to the rc once it's cut. If this makes 5.31 it is 5.31 - if not it's 5.32 - the core release cycle is regular enough that missing a release shouldn't be too big a delay - but if you want to deploy to clients before it is in an official release you can patch their release / add the columns via sql on their release.

Reply:
So we can leave the code as is for now, right?

Reply:
As of right this minute, it should be inside the existing 5_31_alpha1 function. But maybe wait and see if anyone else has comments on the PR? For testing I moved them there and they ran properly.